pulp_python/app/utils.py

```diff
@@ -414,6 +426,53 @@ def write_simple_detail(project_name, project_packages, streamed=False):
     return detail.stream(**context) if streamed else detail.render(**context)
 
 
+def write_simple_index_json(project_names):
+    """Writes the simple index in JSON format."""
+    return {
+        "meta": {"api-version": PYPI_API_VERSION, "_last-serial": PYPI_SERIAL_CONSTANT},
+        "projects": [
+            {"name": name, "_last-serial": PYPI_SERIAL_CONSTANT} for name in project_names
+        ],
+    }
+
+
+def write_simple_detail_json(project_name, project_packages):
+    """Writes the simple detail page in JSON format."""
+    return {
+        "meta": {"api-version": PYPI_API_VERSION, "_last-serial": PYPI_SERIAL_CONSTANT},
+        "name": canonicalize_name(project_name),
+        "files": [
+            {
+                # v1.0, PEP 691
+                "filename": package["filename"],
+                "url": package["url"],
+                "hashes": {"sha256": package["sha256"]},
+                "requires_python": package["requires_python"] or None,
+                # data-dist-info-metadata is deprecated alias for core-metadata
+                "data-dist-info-metadata": (
+                    {"sha256": package["sha256_metadata"]} if package["sha256_metadata"] else False
+                ),
+                "yanked": (
+                    package["yanked_reason"]
+                    if package["yanked"] and package["yanked_reason"]
+                    else package["yanked"]
+                ),
+                # gpg-sig (not in warehouse)
+                # todo (from new PEPs):
+                # size (v1.1, PEP 700)
+                # upload-time (v1.1, PEP 700)
+                # core-metadata (PEP 7.14)
+                # provenance (v1.3, PEP 740)
```

Comment:
Let's leave these for future PRs

Reply:
do you want the comments wholly removed?  I can create new issues for them if they don't already exist